tests/unit/test_yamlhelper.py

```diff
@@ -161,3 +163,22 @@ def test_unroll_yaml_anchors(self):
         )
         actual = yaml_dump(template)
         self.assertEqual(actual, expected)
+
+    def test_unquoted_template_format_version_to_json(self):
+        input_template = (
+            "AWSTemplateFormatVersion: 2010-09-09\n"
+            "Transform: AWS::Serverless-2016-10-31\n"
+            "Resources:\n"
+            "  HelloWorldFunction:\n"
+            "    Type: AWS::Serverless::Function\n"
+            "    Properties:\n"
+            "      Handler: main.handler\n"
+            "      Runtime: python3.7\n"
+            "      CodeUri: .\n"
+            "      Timeout: 600\n"
+        )
+
+        output = yaml_parse(input_template)
+        # Raises a `TypeError` if an unquoted `AWSTemplateFormatVersion` value has been parsed to a
+        # `datetime` object and not a string by `yaml_parse` when using `--use-json` argument.
+        json.dumps(output)
```

Comment:
Nit: Can we validate the expected date is converted to a string? Not throwing an error is probably enough but would like to ensure we are at least asserting the behavior.